h/h_api/bulk_api/command_processor.py

```diff
@@ -175,15 +176,20 @@ def _process_reports(self, data_type, batch, reports):
             if reference is not None:
                 self.id_refs.add_concrete_id(data_type, reference, report.id)
 
-        if self.config.view is not None:
+        if self.config.view is not ViewType.NONE:
             # Store reports for each item, so we ask for the objects to produce
             # our final response
-            self.reports[data_type].append(reports)
+            self.reports[data_type].extend(reports)
 
     def _report_back(self):
-        if not self.reports:
+        if not self.reports or self.config.view is ViewType.NONE:
```

Comment:
As a result of code elsewhere we shouldn't get reports when view type is None, but why rely on that?